src/lib/testing/random.ts

```diff
@@ -843,12 +845,12 @@ function bignatWithInvalid(max: bigint): RandomWithInvalid<bigint> {
   return Object.assign(valid, { invalid });
 }
 
-function fieldWithInvalid(
-  F: typeof Field | typeof Scalar
-): RandomWithInvalid<bigint> {
+function fieldWithInvalid(F: FiniteField): RandomWithInvalid<bigint> {
   let randomField = Random_(F.random);
-  let specialField = oneOf(0n, 1n, F(-1));
-  let field = oneOf<bigint[]>(randomField, randomField, uint64, specialField);
+  let specialField = oneOf(0n, 1n, F.negate(1n));
+  let roughLogSize = 1 << Math.ceil(Math.log2(F.sizeInBits) - 1);
```

Comment:
Why are you using this size?

Reply:
because the `biguint` random generator I'm passing it in needs the bit size to be a power of 2 (for efficiency)